build-system/tasks/report-test-status.js

```diff
@@ -18,70 +18,82 @@
 const argv = require('minimist')(process.argv.slice(2));
 const log = require('fancy-log');
 const requestPromise = require('request-promise');
+const {
+  isPullRequestBuild,
+  isGithubActionsBuild,
+  isTravisBuild,
+} = require('../common/ci');
 const {ciJobUrl} = require('../common/ci');
 const {cyan, green, yellow} = require('ansi-colors');
 const {gitCommitHash} = require('../common/git');
-const {isPullRequestBuild, isTravisBuild} = require('../common/ci');
 
 const reportBaseUrl = 'https://amp-test-status-bot.appspot.com/v0/tests';
 
 const IS_GULP_INTEGRATION = argv._[0] === 'integration';
 const IS_GULP_UNIT = argv._[0] === 'unit';
 const IS_GULP_E2E = argv._[0] === 'e2e';
 
-const IS_LOCAL_CHANGES = !!argv.local_changes;
-const IS_DIST = !!argv.compiled;
-const IS_ESM = !!argv.esm;
-
-const TEST_TYPE_SUBTYPES = new Map([
-  ['integration', ['local', 'minified', 'esm']],
-  ['unit', ['local', 'local-changes']],
-  ['e2e', ['local']],
-]);
+const TEST_TYPE_SUBTYPES = isGithubActionsBuild()
+  ? new Map([
+      ['integration', ['firefox', 'safari', 'edge', 'ie']],
+      ['unit', ['firefox', 'safari', 'edge']],
+    ])
+  : isTravisBuild()
+  ? new Map([
+      ['integration', ['unminified', 'minified', 'esm']],
+      ['unit', ['unminified', 'local-changes']],
+      ['e2e', ['minified']],
+    ])
+  : new Map([]);
 const TEST_TYPE_BUILD_TARGETS = new Map([
   ['integration', ['RUNTIME', 'FLAG_CONFIG', 'INTEGRATION_TEST']],
   ['unit', ['RUNTIME', 'UNIT_TEST']],
   ['e2e', ['RUNTIME', 'FLAG_CONFIG', 'E2E_TEST']],
 ]);
 
 function inferTestType() {
-  if (IS_GULP_E2E) {
-    return 'e2e/local';
-  }
-
-  let type;
-  if (IS_GULP_UNIT) {
-    type = 'unit';
-  } else if (IS_GULP_INTEGRATION) {
-    type = 'integration';
-  } else {
+  // Determine type (early exit if there's no match).
+  const type = IS_GULP_E2E
+    ? 'e2e'
+    : IS_GULP_INTEGRATION
+    ? 'integration'
+    : IS_GULP_UNIT
+    ? 'unit'
+    : null;
+  if (type == null) {
     return null;
   }
 
-  if (IS_LOCAL_CHANGES) {
-    return `${type}/local-changes`;
-  } else if (IS_DIST) {
-    if (IS_ESM) {
-      return `${type}/esm`;
-    }
-    return `${type}/minified`;
-  } else {
-    return `${type}/local`;
-  }
+  // Determine subtype (more specific cases come first).
+  const subtype = argv.local_changes
+    ? 'local-changes'
+    : argv.esm
+    ? 'esm'
+    : argv.firefox
+    ? 'firefox'
+    : argv.safari
+    ? 'safari'
+    : argv.edge
+    ? 'edge'
+    : argv.ie
+    ? 'ie'
+    : argv.compiled
+    ? 'minified'
+    : 'unminified';
+
+  return `${type}/${subtype}`;
 }
 
 async function postReport(type, action) {
-  // TODO(rsimha, ampproject/amp-github-apps#1111) Remove Travis special-case.
-  if (type !== null && isTravisBuild() && isPullRequestBuild()) {
+  if (type !== null && isPullRequestBuild()) {
```

Comment:
I wonder if we can use `type ?? isPullRequestBuild()` by now :)

Reply:
Good point, we are indeed using node v14! In this case, since the expression's type is intended to be boolean and not string, I simplified this to logical and instead of nullish coalescing.